### ISSUE TITLE
Plotting of instruction schedule from mock backend fails

### ISSUE THREAD

**Issue by peendebak** [Qiskit/qiskit-terra#5951]
### Information

- **Qiskit Terra version**: '0.17.0.dev0+6f0eca6'
- **Python version**: 3.8.6
- **Operating system**: Windows 10

The output of `qiskit.version.QiskitVersion()` is
```
{'qiskit-terra': '0.17.0.dev0+6f0eca6', 'qiskit-aer': '0.7.3', 'qiskit-ignis': '0.5.1', 'qiskit-ibmq-provider': '0.11.1', 'qiskit-aqua': '0.8.1', 'qiskit': '0.23.4'}
```

### What is the current behavior?

When importing the `FakeAthens`, gettting the schedule for the `u3` instruction and plotting an exception is generated.

Depending on this qiskit version I get different error messages. The version reported above yields:
```
...
  File "c:\projects\qiskit-terra\qiskit\visualization\pulse_v2\generators\frame.py", line 318, in gen_frame_symbol
    program.append('{}({})'.format(inst.__class__.__name__, inst.phase.name))

AttributeError: 'ParameterExpression' object has no attribute 'name'
```

### Steps to reproduce the problem

A minimal example is:
```
from qiskit.test.mock import FakeAthens

backend = FakeAthens()
defaults = backend.defaults()
inst_map = defaults.instruction_schedule_map
schedule=inst_map.get('u3', [0])
schedule.draw()
```
print(qiskit.version.QiskitVersion())

### What is the expected behavior?

The schedule should be plotted

### Suggested solutions

None

**Comment by taalexander** [comment]
@nkanazawa1989 I believe this was a bug introduced with the release of parameters + schedules. Would you mind having a look?

**Comment by peendebak** [comment]
@taalexander @nkanazawa1989 The issues still exists on current master.
```
In [5]: print(qiskit.version.QiskitVersion())
{'qiskit-terra': '0.18.0.dev0+98130dd', 'qiskit-aer': '0.8.0', 'qiskit-ignis': '0.6.0', 'qiskit-ibmq-provider': '0.12.2', 'qiskit-aqua': '0.9.0', 'qiskit': '0.25.0', 'qiskit-nature': None, 'qiskit-finance': None, 'qiskit-optimization': None, 'qiskit-machine-learning': None}
```

**Comment by taalexander** [comment]
Investigating this a result of the implementation of `partial` which does not fully bind your argument but rather creates a new function with the partially bound argument as a keyword argument. 

```python
import inspect

inspect.signature(partial(callable_schedule2, amplitude=.4))
```
```
>>> <Signature (f, *, amplitude=0.4)>
```

See [partial docs](https://docs.python.org/3/library/functools.html#functools.partial) for a description of how it works. In general I would expect any function with keyword arguments to have this issue as these keyword arguments would be treated as arguments for the instruction calibration, we do not support optional arguments in mapping circuit instructions to schedules as circuit instructions do not support optional arguments themselves.

Its a question of what the expected behaviour should be? The two options are:
1. Keyword arguments are mandatory arguments for the schedule generation method from the `InstructionScheduleMap`'s perspective.
2. Keyword arguments are ignored from the instruction schedule maps perspective.

**Comment by nkanazawa1989** [comment]
Thanks for reporting the bug. Fixed in #6282. I just mistook the data type of `.phase`. This can be `ParameterExpression` (I assumed `Parameter`) which doesn't have `.name` property.
